core/services/relay/evm/mercury/orm.go

```diff
@@ -0,0 +1,114 @@
+package mercury
+
+import (
+	"context"
+	"crypto/sha256"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/smartcontractkit/sqlx"
+
+	ocrtypes "github.com/smartcontractkit/libocr/offchainreporting2plus/types"
+
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services/pg"
+	"github.com/smartcontractkit/chainlink/v2/core/services/relay/evm/mercury/wsrpc/pb"
+)
+
+type ORM struct {
+	q pg.Q
+}
+
+func NewORM(db *sqlx.DB, lggr logger.Logger, cfg pg.QConfig) *ORM {
+	namedLogger := lggr.Named("MercuryORM")
+	q := pg.NewQ(db, namedLogger, cfg)
+	return &ORM{
+		q: q,
+	}
+}
+
+// InsertTransmitRequest inserts one transmit request if the payload does not exist already.
+func (o *ORM) InsertTransmitRequest(ctx context.Context, req *pb.TransmitRequest, reportCtx ocrtypes.ReportContext, qopts ...pg.QOpt) error {
+	q := o.q.WithOpts(append([]pg.QOpt{pg.WithParentCtx(ctx)}, qopts...)...)
```

Comment:
Generally speaking if it takes `qopts...` you can omit the first `ctx` arg and pass it in from the caller like this:

`o.InsertTransmitRequest(req, reportCtx, pg.WithParentCtx(ctx))`